apiserver/charms/client.go

```diff
@@ -16,17 +17,19 @@ import (
 	"github.com/juju/juju/state"
 )
 
-var getState = func(st *state.State) charmsAccess {
-	return stateShim{st}
-}
-
 // Charms defines the methods on the charms API end point.
 type Charms interface {
 	List(args params.CharmsList) (params.CharmsListResult, error)
 	CharmInfo(args params.CharmURL) (params.CharmInfo, error)
 	IsMetered(args params.CharmURL) (bool, error)
 }
 
+type charmsAccess interface {
```

Comment:
Could you renaming this "backend" while you're at it? That's more consistent with other (newer) facades.